crates/modelardb_storage/src/lib.rs

```diff
@@ -446,4 +707,172 @@ mod tests {
+
+        assert_eq!(record_batch, bytes_record_batch);
+    }
+
+    #[test]
+    fn test_convert_invalid_bytes_to_record_batch() {
+        let result = try_convert_bytes_to_record_batch(
+            vec![1, 2, 4, 8],
+            &Arc::new(test::normal_table_schema()),
+        );
+
+        assert_eq!(
+            result.unwrap_err().to_string(),
+            "Arrow Error: Io error: failed to fill whole buffer"
+        );
+    }
+
+    #[test]
+    fn test_convert_bytes_with_invalid_schema_to_record_batch() {
+        let bytes = try_convert_record_batch_to_bytes(&test::normal_table_record_batch()).unwrap();
+
+        let field = Field::new("field", ArrowValue::DATA_TYPE, false);
+        let schema = Arc::new(Schema::new(vec![field]));
+        let result = try_convert_bytes_to_record_batch(bytes, &schema);
+
+        assert_eq!(
+            result.unwrap_err().to_string(),
+            "Arrow Error: Invalid argument error: column types must match schema types, expected \
+            Float32 but found Timestamp(Microsecond, None) at column index 0"
+        );
+    }
+
+    // Tests for normal_table_metadata_record_batch() and model_table_metadata_record_batch().
+    #[test]
+    fn test_normal_table_metadata_record_batch() {
+        let schema = test::normal_table_schema();
+        let record_batch =
+            normal_table_metadata_record_batch(test::NORMAL_TABLE_NAME, &schema).unwrap();
+
+        assert_eq!(**record_batch.column(0), StringArray::from(vec!["normal"]));
+        assert_eq!(
+            **record_batch.column(1),
+            StringArray::from(vec![test::NORMAL_TABLE_NAME])
+        );
+        assert_eq!(
+            **record_batch.column(2),
+            BinaryArray::from_vec(vec![&try_convert_schema_to_bytes(&schema).unwrap()])
+        );
+    }
+
+    #[test]
+    fn test_model_table_metadata_record_batch() {
+        let model_table_metadata = test::model_table_metadata();
+        let record_batch = model_table_metadata_record_batch(&model_table_metadata).unwrap();
+
+        assert_eq!(**record_batch.column(0), StringArray::from(vec!["model"]));
+        assert_eq!(
+            **record_batch.column(1),
+            StringArray::from(vec![test::MODEL_TABLE_NAME])
+        );
+
+        let expected_schema_bytes =
+            try_convert_schema_to_bytes(&model_table_metadata.query_schema).unwrap();
+        assert_eq!(
+            **record_batch.column(2),
+            BinaryArray::from_vec(vec![&expected_schema_bytes])
+        );
+
+        let error_bounds_array = modelardb_types::array!(record_batch, 3, ListArray).value(0);
+        let value_array = error_bounds_array
+            .as_any()
+            .downcast_ref::<Float32Array>()
+            .unwrap();
+
+        assert_eq!(value_array, &Float32Array::from(vec![0.0, 1.0, -5.0, -0.0]));
+
+        let generated_columns_array = modelardb_types::array!(record_batch, 4, ListArray).value(0);
+        let expr_array = generated_columns_array
```

Comment:
Maybe use the `array!()` macro to shorten the code and make it easy to return the error in the future for casts by changing the macro.

Reply:
`array!()` is used to extract a column from a record batch which is not the case here. Since this pattern is only used a few times in this file, I have not added a new macro for this specific case.

Reply:
Ups, for some reason I thought of `error_bounds_array` as `RecordBatch`.